### PR TITLE
8272778: Consolidate is_instance and is_instance_inlined in java_lang_String

### PR DESCRIPTION
**Pull request by albertnetymk**
Simple change of merging `is_instance` and `is_instance_inlined`.

Test: build

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8272778](https://bugs.openjdk.java.net/browse/JDK-8272778): Consolidate is_instance and is_instance_inlined in java_lang_String

### Reviewers
 * [Coleen Phillimore](https://openjdk.java.net/census#coleenp) (@coleenp - **Reviewer**)
 * [Claes Redestad](https://openjdk.java.net/census#redestad) (@cl4es - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5201/head:pull/5201` \
`$ git checkout pull/5201`

Update a local copy of the PR: \
`$ git checkout pull/5201` \
`$ git pull https://git.openjdk.java.net/jdk pull/5201/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5201`

View PR using the GUI difftool: \
`$ git pr show -t 5201`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5201.diff">https://git.openjdk.java.net/jdk/pull/5201.diff</a>

</details>

### CODE REVIEW

**Comment by cl4es** [comment]
It seems the split between `is_instance` and `is_instance_inlined` was introduced recently (5314d28f8412e9c993880884a3b626f05b590743) and I the point was likely to actively opt-out of inlining in performance-insensitive places (also makes the generated code a bit less noisy and easier to debug, by some accounts). 

Since it was done intentionally I'd check we don't lose something by consolidating these into a single inlined version. E.g. measure that the static size of libjvm doesn't grow by much.

File: src/hotspot/share/classfile/javaClasses.inline.hpp [hunk 1]
```diff
@@ -123,7 +123,7 @@ int java_lang_String::length(oop java_string) {
   return length(java_string, value);
 }
 
-bool java_lang_String::is_instance_inlined(oop obj) {
+bool java_lang_String::is_instance(oop obj) {
```
Comment (coleenp) [inline src/hotspot/share/classfile/javaClasses.inline.hpp:126]: I don't really see anything wrong with this.  is_instance() is called above in asserts but that shouldn't affect performance by increasing the size of generated code.

**Comment by albertnetymk** [comment]
> It seems the split between is_instance and is_instance_inlined was introduced recently (5314d28)

I believe the split was introduced in [8072911: Remove includes of oop.inline.hpp from .hpp files](https://github.com/openjdk/jdk/commit/4913ad5d7dcc8ae4e034aa110a97ea8313b2b3ee) to address potential circular dependencies.

> measure that the static size of libjvm doesn't grow by much.

Before:

```
ls -l build/*/images/jdk/lib/server/libjvm.so
-rw-r--r-- 1 albert albert 213664560 Aug 20 19:01 build/linux-x64-debug/images/jdk/lib/server/libjvm.so
-rw-r--r-- 1 albert albert  24243456 Aug 20 19:11 build/linux-x64-release/images/jdk/lib/server/libjvm.so
```

After:

```
ls -l build/*/images/jdk/lib/server/libjvm.so
-rw-r--r-- 1 albert albert 213727136 Aug 20 19:12 build/linux-x64-debug/images/jdk/lib/server/libjvm.so
-rw-r--r-- 1 albert albert  24247552 Aug 20 19:12 build/linux-x64-release/images/jdk/lib/server/libjvm.so
```

In summary:

for fastdebug: 213664560 -> 213727136, +62576 bytes, +0.03%
for release: 24243456 -> 24247552, +4096 bytes, +0.02%

File: src/hotspot/share/classfile/javaClasses.inline.hpp (hunk 1 above)
Comment (albertnetymk) [reply src/hotspot/share/classfile/javaClasses.inline.hpp:126]: `is_instance()` is called in non-assertion code as well, e.g. `ConstantPool::add_dumped_interned_strings`. By merging the two versions, one wouldn't use the potentially slower version by accident, and it's also more consistent with `is_instance` of other classes (e.g. `java_lang_Class`) in the same file.

Comment (cl4es) [reply src/hotspot/share/classfile/javaClasses.inline.hpp:126]: While `ConstantPool::add_dumped_interned_strings` might be an example of something that's not performance sensitive (even with dynamic CDS), the 0.02%/4Kb static size increase is less than concerning. Thanks for checking!

**Review by coleenp — APPROVED** [review]
Looks good! Thanks for checking for the non-inlined version of is_instance.

**Review by cl4es — APPROVED** [review]
FWIW this looks good to me too. 👍🏽

**Comment by albertnetymk** [comment]
More renaming because of new code from master branch.

**Review by cl4es — APPROVED** [review]

**Review by coleenp — APPROVED** [review]
Ok.

**Comment by albertnetymk** [comment]
Thank you for the review.

/integrate

**Pull request closed without merging**